nearcore/src/config.rs

```diff
@@ -1326,10 +1372,85 @@ impl From<NodeKeyFile> for KeyFile {
+    };
+    let mut validation_errors = ValidationErrors::new();
+    // if not able to create Config from config.json, the program will stop. The error is already pushed to validation_errors.
+    let config = Config::from_file_no_panic(&dir.join(CONFIG_FILENAME), &mut validation_errors)?;
+    let genesis_file = dir.join(&config.genesis_file);
+    let validator_file = dir.join(&config.validator_key_file);
+    let validator_signer = if validator_file.exists() {
+        match InMemoryValidatorSigner::from_file(&validator_file) {
+            Ok(signer) => Some(Arc::new(signer) as Arc<dyn ValidatorSigner>),
+            Err(_) => None,
+        }
+    } else {
+        let error_message =
+            format!("validator key file does not exist at the path {}", validator_file.display());
+        validation_errors
+            .push_errors(ValidationError::ValidatorKeyFileError { error_message: error_message });
+        None
+    };
+    let node_key_path = dir.join(&config.node_key_file);
+    let network_signer_result = NodeKeyFile::from_file(&node_key_path);
+    let network_signer = match network_signer_result {
+        Ok(node_key_file) => Some(node_key_file),
+        Err(_) => {
+            validation_errors.push_errors(ValidationError::NodeKeyFileError {
+                error_message: format!(
+                    "Failed reading node key file from {}",
+                    node_key_path.display()
+                ),
+            });
+            None
+        }
+    };
+
+    let genesis_result = match &config.genesis_records_file {
+        Some(records_file) => Genesis::from_files_no_panic(
+            &genesis_file,
+            &dir.join(records_file),
+            genesis_validation,
+            &mut validation_errors,
+        ),
+        None => {
+            Genesis::from_file_no_panic(&genesis_file, genesis_validation, &mut validation_errors)
+        }
+    };
+
+    let genesis = match genesis_result {
+        Ok(genesis) => Some(genesis),
+        Err(_) => None,
+    };
+
+    validation_errors.panic_if_errors();
+
+    if genesis.is_none() || network_signer.is_none() {
+        panic!("Genesis and network_signer should not be None by now.")
+    }
+    let near_config = NearConfig::new(
+        config,
+        genesis.unwrap(),
+        network_signer.unwrap().into(),
+        validator_signer,
+    )?;
+    Ok(near_config)
+}
+
 pub fn load_config(
     dir: &Path,
     genesis_validation: GenesisValidationMode,
 ) -> anyhow::Result<NearConfig> {
+    let genesis_validation = match genesis_validation {
+        GenesisValidationMode::Full => GenesisValidationMode::Full,
+        GenesisValidationMode::UnsafeFast => GenesisValidationMode::UnsafeFast,
+    };
     let config = Config::from_file(&dir.join(CONFIG_FILENAME))?;
     let genesis_file = dir.join(&config.genesis_file);
     let validator_file = dir.join(&config.validator_key_file);
```

Comment:
@nikurt question: since in our code, the location of genesis_file, node_key_file and validator_file all depends on corresponding fields in config.json, would it be fine to directly panic whenever config.json has file issue or fail semantic checks? Once config.json is fixed, all errors in genesis.json, validator_key.json and node_key.json will be reported altogether in one panic.